src/topNavigation/topNavigation.tsx

```diff
@@ -31,139 +38,153 @@ interface State {
-  );
-  const matchFavourites = useMatch(getRouteById(AppRoutes.FAVOURITES));
-  const changeLanguage = (language: Language) => {
-    i18n.changeLanguage(language).then(() => {
-      document.location.reload();
-    });
+  const naviLinks: TopNavigationLinkProps[] = [
+    {
+      to: getRouteById(AppRoutes.PLOT_SEARCH_AND_COMPETITIONS),
+      label: t(
+        'topNavigation.tabs.plotSearchAndCompetitions',
+        'Plot search and competitions',
+      ),
+    },
+    {
+      to: getRouteById(AppRoutes.OTHER_COMPETITIONS_AND_SEARCHES),
+      label: t(
+        'topNavigation.tabs.otherCompetitionsAndSearches',
+        'Other competitions and searches',
+      ),
+    },
+    {
+      to: getRouteById(AppRoutes.AREA_SEARCH_LANDING),
+      label: t('topNavigation.tabs.areaSearch', 'Area search'),
+    },
+  ];
+
+  const languages: LanguageOption[] = [
+    {
+      label: t('language.fi', 'Finnish'),
+      value: Language.FI,
+      isPrimary: false,
+    },
+    {
+      label: t('language.se', 'Swedish'),
+      value: Language.SV,
+      isPrimary: false,
+    },
+    {
+      label: t('language.en', 'English'),
+      value: Language.EN,
+      isPrimary: false,
+    },
+  ];
+
+  const changeLanguage = (lang: string) => {
+    if (lang !== i18n.language) {
+      i18n.changeLanguage(lang).then(() => {
+        document.location.reload();
+      });
+    }
   };
 
   return (
-    <Navigation
-      menuToggleAriaLabel={t('topNavigation.menuLabel', 'Menu')}
-      skipTo="#content"
-      skipToContentLabel={t('topNavigation.skipToContent', 'Skip to content')}
-      onTitleClick={() => returnHome(navigate)}
-      fixed
+    <Header
+      onDidChangeLanguage={(lang) => changeLanguage(lang)}
+      languages={languages}
+      defaultLanguage={i18n.language}
       className="TopNavigation"
-      logoLanguage={i18n.language as LogoLanguage}
     >
-      <Navigation.Row variant="inline">
-        <Navigation.Item
-          label={t(
-            'topNavigation.tabs.plotSearchAndCompetitions',
-            'Plot search and competitions'
-          )}
-          as={Link}
-          to={getRouteById(AppRoutes.PLOT_SEARCH_AND_COMPETITIONS)}
-          active={matchPlotSearchAndCompetitions !== null}
+      <Header.SkipLink
+        skipTo="#content"
+        label={t('topNavigation.skipToContent', 'Skip to content')}
+      />
+      <Header.ActionBar
+        title={t(
+          'mainAppTitle',
+          'City of Helsinki plot and land leasing system',
+        )}
+        frontPageLabel={'etusivu'}
+        titleAriaLabel={'hedertitteliarialable'}
+        titleHref={getRouteById(AppRoutes.HOME)}
+        logo={<Logo src={logoFi} alt={'altti'} />}
+        logoAriaLabel={'logoAriaLabel'}
+        logoHref={getRouteById(AppRoutes.HOME)}
+        menuButtonAriaLabel={t('language.languageSelection', 'Language')}
+        openFrontPageLinksAriaLabel="Avaa Etusivun linkkivalikko"
+      >
+        <Header.LanguageSelector
```

Comment:
An oddity I found: I can only get the menu to open by clicking the language shortcode, not the arrow button next to it. It seems to work as expected in the HDS storybook, but I don't see any obvious reason why it should work differently in your implementation.

Reply:
Yup. This is weird. Maybe this could be reported to HDS.